Userland/Libraries/LibWeb/Infra/Base64.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022-2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Array.h>
+#include <AK/ByteBuffer.h>
+#include <AK/CharacterTypes.h>
+#include <AK/Error.h>
+#include <AK/StringBuilder.h>
+#include <AK/StringView.h>
+#include <AK/Vector.h>
+#include <LibWeb/Infra/Base64.h>
+
+namespace Web::Infra {
+
+static constexpr Array base64_alphabet = {
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H',
+    'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P',
+    'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
+    'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n',
+    'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+    'w', 'x', 'y', 'z', '0', '1', '2', '3',
+    '4', '5', '6', '7', '8', '9', '+', '/'
+};
+
+static consteval auto make_base64_lookup_table()
+{
+    Array<i16, 256> table;
+    table.fill(-1);
+    for (size_t i = 0; i < base64_alphabet.size(); ++i) {
+        table[base64_alphabet[i]] = static_cast<i16>(i);
+    }
+    return table;
+}
+
+static constexpr auto base64_alphabet_lookup_table = make_base64_lookup_table();
+
+// https://infra.spec.whatwg.org/#forgiving-base64
+ErrorOr<ByteBuffer> decode_forgiving_base64(StringView input)
+{
+    // 1. Remove all ASCII whitespace from data.
+    // FIXME: It is possible to avoid copying input here, it's just a bit tricky to remove the equal signs
+    StringBuilder builder;
+    for (auto character : input) {
+        if (!is_ascii_space(character))
```

Comment:
Please put the commit moving it first and then use the proper definition of ASCII whitespace in the fix